### PR TITLE
Updated addon docs, removed deprecated Viewport component

### PR DESCRIPTION
**Pull request by mizx**
## What I did
I went through most of the `@storybook/addon-*` libs to update/add DefinitelyTyped/TypeScript definition files. Along the way I noticed a few typos or items being deprecated and still mentioned in the 4.0 docs.

## How to test
Check out my PR :)

### CODE REVIEW

**Review by danielduan — APPROVED** [review]

**Review by Keraito — APPROVED** [review]

**Comment by danielduan** [comment]
I have no idea what's going on with the teamcity CI but this is ready to be merged.

**Comment by igor-dv** [comment]
merge from master should help

**Comment by Hypnosphi** [comment]
@danielduan why `patch`? The deprecation was made in `4.0`

File: addons/actions/README.md [hunk 1]
```diff
@@ -134,11 +134,11 @@ import { storiesOf } from '@storybook/html';
 import { withActions } from '@storybook/addon-actions';
 
 storiesOf('button', module)
-  // Log mousovers on entire story and clicks on .btn
+  // Log mouseovers on entire story and clicks on .btn
   .addDecorator(withActions('mouseover', 'click .btn'))
   .add('with actions', () => `
     <div>
-      Clicks on this button will be logged: <button class="btn" type="button">Button</button>
+      Clicks on this button will be logged: <button className="btn" type="button">Button</button>
```
Comment (Hypnosphi) [inline addons/actions/README.md]: This is HTML not JSX. Please change it back to `class`

**Pull request merged**